pkg/debug/info.go

```diff
@@ -0,0 +1,140 @@
+	tempDir := os.TempDir()
+	debugFilePrefix := "odo-debug.json"
+	s := []string{client.Namespace, appName, componentName, debugFilePrefix}
+	debugFileName := strings.Join(s, "-")
+	return filepath.Join(tempDir, debugFileName)
+}
+
+func CreateDebugInfoFile(f *DefaultPortForwarder, portPair string) error {
+	return createDebugInfoFile(f, portPair, filesystem.DefaultFs{})
+}
+
+// createDebugInfoFile creates a file in the temp directory with information regarding the debugging session of a component
+func createDebugInfoFile(f *DefaultPortForwarder, portPair string, fs filesystem.Filesystem) error {
+	portPairs := strings.Split(portPair, ":")
+	if len(portPairs) > 2 || len(portPairs) < 2 {
+		return errors.New("port pair should be of the format localPort:RemotePair")
+	}
+
+	localPort, err := strconv.Atoi(portPairs[0])
+	if err != nil {
+		return errors.New("local port should be a int")
+	}
+	remotePort, err := strconv.Atoi(portPairs[1])
+	if err != nil {
+		return errors.New("remote port should be a int")
+	}
+
+	odoDebugFile := OdoDebugFile{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "OdoDebugInfo",
+			APIVersion: "v1",
+		},
+		DebugProcessId: os.Getpid(),
+		ProjectName:    f.client.Namespace,
+		AppName:        f.appName,
+		ComponentName:  f.componentName,
+		RemotePort:     remotePort,
+		LocalPort:      localPort,
+	}
+	odoDebugPathData, err := json.Marshal(odoDebugFile)
+	if err != nil {
+		return errors.New("error marshalling json data")
+	}
+
+	// writes the data to the debug info file
+	file, err := fs.OpenFile(GetDebugInfoFilePath(f.client, f.componentName, f.appName), os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
+	if err != nil {
+		return err
+	}
+	_, err = file.Write(odoDebugPathData)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func GetDebugInfo(f *DefaultPortForwarder) (OdoDebugFile, bool) {
+	return getDebugInfo(f, filesystem.DefaultFs{})
+}
+
+// getDebugInfo gets information regarding the debugging session of the component
+// returns the OdoDebugFile from the debug info file
+// returns true if debugging is running else false
+func getDebugInfo(f *DefaultPortForwarder, fs filesystem.Filesystem) (OdoDebugFile, bool) {
+	// gets the debug info file path and reads/unmarshals it
+	debugInfoFilePath := GetDebugInfoFilePath(f.client, f.componentName, f.appName)
+	readFile, err := fs.ReadFile(debugInfoFilePath)
+	if err != nil {
+		glog.V(4).Infof("the debug %v is not present", debugInfoFilePath)
+		return OdoDebugFile{}, false
+	}
+
+	var odoDebugFileData OdoDebugFile
+	err = json.Unmarshal(readFile, &odoDebugFileData)
+	if err != nil {
+		glog.V(4).Infof("couldn't unmarshal the debug file %v", debugInfoFilePath)
+		return OdoDebugFile{}, false
+	}
+
+	// get the debug process id and send a signal 0 to check if it's alive or not
+	processInfo, err := os.FindProcess(odoDebugFileData.DebugProcessId)
+	if err != nil || processInfo == nil {
```

Comment:
according to the docs of `FindProcess` https://golang.org/pkg/os/#FindProcess that
```
On Unix systems, FindProcess always succeeds and returns a Process for the given pid, regardless of whether the process exists.
```
so would this ever trigger?

Reply:
added a comment regarding this